src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs

```diff
@@ -1,21 +1,147 @@
 ﻿namespace Dfe.Spi.RatesAdapter.Application
 {
+    using System;
     using System.Threading;
     using System.Threading.Tasks;
     using Dfe.Spi.Models.Entities;
+    using Dfe.Spi.Models.RatesModels;
     using Dfe.Spi.RatesAdapter.Application.Definitions;
+    using Dfe.Spi.RatesAdapter.Domain.Definitions;
+    using LocalDomainModels = Dfe.Spi.RatesAdapter.Domain.Models;
 
     /// <summary>
     /// Implements <see cref="IRatesManager" />.
     /// </summary>
     public class RatesManager : IRatesManager
     {
+        private readonly ISchoolInformationStorageAdapter schoolInformationStorageAdapter;
+
+        /// <summary>
+        /// Initialises a new instance of the <see cref="RatesManager" />
+        /// class.
+        /// </summary>
+        /// <param name="schoolInformationStorageAdapter">
+        /// An instance of type
+        /// <see cref="ISchoolInformationStorageAdapter" />.
+        /// </param>
+        public RatesManager(
+            ISchoolInformationStorageAdapter schoolInformationStorageAdapter)
+        {
+            this.schoolInformationStorageAdapter = schoolInformationStorageAdapter;
+        }
+
         /// <inheritdoc />
-        public Task<Rates> GetRatesAsync(
-            string id,
+        public async Task<Rates> GetRatesAsync(
+            int year,
+            string entityName,
+            string identifier,
             CancellationToken cancellationToken)
         {
-            throw new System.NotImplementedException();
+            Rates toReturn = null;
+
+            if (entityName == nameof(LearningProvider))
+            {
+                long urn;
+                if (!long.TryParse(identifier, out urn))
+                {
+                    // TODO: Throw some sort of exception.
+                }
+
+                LocalDomainModels.SchoolInformation schoolInformation =
+                    await this.schoolInformationStorageAdapter.GetSchoolInformationAsync(
+                        year,
+                        urn,
+                        cancellationToken)
+                        .ConfigureAwait(false);
+
+                toReturn = Map(schoolInformation);
+            }
+            else
+            {
+                throw new NotImplementedException(
+                    "Not yet done! Implement me!");
+            }
+
+            return toReturn;
+        }
+
+        private static Rates Map(
```

Comment:
As you have AutoMapper already referenced in the solution, would it not be easier to use it here too?

Reply:
The reason I've not is, the spi-models lib doesn't implement an interface (unlike in the infrastructure/domain), so they're not guarenteed to be the same.

This at least lets us know at compilation time after upgrading the package that something isn't mapping correctly. Looks messy, I know, but it'll be easier to manage!